### ISSUE TITLE
inconsistency regarding supported QUIC versions

### ISSUE THREAD

**Issue by marten-seemann** [tfpauly/quic-proxy#83]
The Register Target Connection ID capsule and the Virtual Connection ID capsule allow connection IDs of up to 255 bytes. This is because RFC 8999 allows up to 255 bytes, while RFC 9000 restricts them to 20 bytes.

The capsules also contain stateless reset tokens. However, stateless resets are not a concept defined in RFC 8999, they’re only defined in RFC 9000.

It seems like the hope when defining this extension is that once a new QUIC version that allows longer CIDs ships, it will still contain the concept of stateless reset tokens, with unchanged semantics and wire format. This might or might not hold true.

**Comment by tfpauly** [comment]
That's a good point, but I don't think we should make any change.

- If there is a new version that has the same CID lengths and reset token behavior as v1, we're good.
- If there is a new version that has longer CID length, but the same reset token behavior as v1, we're good.
- If there is a new version that has no reset token support, then we can just leave those as empty, and we're still good.
- If there is a new version that needs entirely different reset token semantics or format, then we define new capsule types when we get there for that version.

Would you agree?

**Comment by marten-seemann** [comment]
I agree with your summary. Should this document specify that the extension only works with QUIC versions that satisfy (1) - (3), but not (4)?

**Comment by tfpauly** [comment]
Yes, I think it would be good to add a line to explain that if the behavior of stateless reset changes or other requirements are added, we need new capsules.

**Comment by tfpauly** [comment]
Closed by #139
